src/printf.c

```diff
@@ -298,15 +298,15 @@ UNUSED s32 func_800B63F4(const char *format, ...) {
     s32 fontCharU;
     s32 stringLength;
     char s[255];
-    char *ch;
+    u8 *ch;
     va_list args;
     va_start(args, format);
 
     stringLength = 0;
     func_800B4A08(1);
     vsprintf(s, format, args);
     func_800B4A08(0);
-    for (ch = &s[0]; *ch != '\0'; ch++) {
+    for (ch = (u8 *) &s[0]; *ch != '\0'; ch++) {
```

Comment:
This change shouldn't be necessary since char is already unsigned. It's maybe arbitrary, but I tend to use char when I know I'm dealing with text, and u8 when it's being treated as arbitrary bytes.

Reply:
Not casting throws a warning, GCC gets picky between u8 and char, even if they expand to exactly the same thing. The usage of this var is also that of a u8 in an array.